### PR TITLE
ECB shift fixes

### PR DESCRIPTION
**Pull request by jobrien97**
- Our ECB shifts now properly function identically to Melee/PM
   - After 10 frames of airtime, the bottom point of your ECB now shifts upwards to your character's knees
   - Previously, we shifted the entire ECB upwards, which would cause your character's head collision to be raised as well
- Our old ECB shift implementation unknowingly raised ledgegrab boxes after 10f of airtime as well, causing unintended side effects
   - This has been fixed; ledgegrab boxes now stay in a consistent position
- ECB shifts now apply during knockback

### CODE REVIEW

File: fighters/common/src/function_hooks/collision.rs [hunk 1]
```diff
@@ -98,10 +98,31 @@ unsafe fn get_touch_flag_hook(boma: &mut BattleObjectModuleAccessor) -> i32 {
     original!()(boma)
 }
 
+#[skyline::hook(offset = 0x6ca950)]
+unsafe fn ground_module_update_hook(ground_module: u64) {
+    call_original!(ground_module);
+    let boma = *((ground_module + 0x20) as *mut *mut BattleObjectModuleAccessor);
+    let line = *((ground_module + 0x28) as *mut *mut f32);
+    let shift = VarModule::get_float((*boma).object(), vars::common::instance::ECB_Y_OFFSETS);
```
Comment (WuBoytH) [inline fighters/common/src/function_hooks/collision.rs]: This VarModule call should be within the `is_fighter` if block
```
if (*boma).is_fighter() {
  let shift = ...;
  if !(*boma).is_status_one_of... {
  }
}
```

**Review by WuBoytH — CHANGES_REQUESTED** [review]

File: fighters/common/src/function_hooks/collision.rs (hunk 1 above)
Comment (jobrien97) [reply fighters/common/src/function_hooks/collision.rs]: Any reason? Will calling VarModule for a non-fighter boma cause potential issues?

Comment (WuBoytH) [reply fighters/common/src/function_hooks/collision.rs]: well if something is passed into this and it *doesn't* have a VarModule, the game will panic, so it's really just to make sure...

Comment (jobrien97) [reply fighters/common/src/function_hooks/collision.rs]: fwiw, i've been playing w this for weeks now, and haven't gotten a panic. But for sanity's sake i can change it, sure

**Comment by stingers135** [comment]
I tested it for about 30 minutes; everything felt fluid and didn't notice any bugs or glitches with ledgegrabbing or with clipping into stage. Spent about 10min straight trying to see if I could get clips to happen from offstage and everything worked as expected.

**Comment by yoonikoo** [comment]
I've tested these changes with Fox, Falco, Wolf, Greninja, and Mewtwo. There really hasn't been anything out of the ordinary to report. Falco and Fox benefit from these changes by allowing to grab ledge properly after an Up-B stall from ledge. LGTM.

**Review by WuBoytH — APPROVED** [review]

**Pull request merged**